src/bun.js/api/bun/socket.zig

```diff
@@ -1352,33 +1350,19 @@ fn NewSocket(comptime ssl: bool) type {
 
             const args = callframe.arguments(4);
 
-            if (args.len == 0) {
-                log("end()", .{});
-                if (!this.detached) {
-                    if (!this.socket.isClosed()) this.socket.flush();
-                    this.detached = true;
-
-                    if (!this.socket.isClosed())
-                        this.socket.close(0, null);
-                    this.markInactive();
-                }
-
-                return JSValue.jsUndefined();
-            }
-
             log("end({d} args)", .{args.len});
 
             if (this.detached) {
                 return JSValue.jsNumber(@as(i32, -1));
             }
 
             const result = this.writeOrEnd(globalObject, args.ptr[0..args.len], true);
-            if (result != .zero and result.toInt32() > 0) {
+            if (result != .zero and result.toInt32() >= 0) {
```

Comment:
0 is fine 

0 means “I have backpressure. I can’t write right now”

Reply:
Ah okay, so we need to differentiate between writing zero-length strings and buffers and this back-pressure case then, since otherwise the former cases is causing leakage.

Reply:
it's actually more complicated than that &minus; if we wrote only part of the buffer in `.end()`, we really shouldn't be closing out the socket just yet - addressed in https://github.com/oven-sh/bun/pull/1700/commits/aa8d4feb71ac49c48190bc11d2ec91487491065b